MatrixSDK/JSONModels/MXJSONModels.m

```diff
@@ -196,13 +197,51 @@ + (id)modelFromJSON:(NSDictionary *)JSONDictionary
         MXJSONModelSetArray(authSession.completed, JSONDictionary[@"completed"]);
         MXJSONModelSetString(authSession.session, JSONDictionary[@"session"]);
         MXJSONModelSetDictionary(authSession.params, JSONDictionary[@"params"]);
+                                
+        NSArray *flows;
+        MXJSONModelSetArray(flows, JSONDictionary[@"flows"]);
         
-        authSession.flows = [MXLoginFlow modelsFromJSON:JSONDictionary[@"flows"]];
+        authSession.flows = [self loginFlowsFromJSON:flows];
     }
     
     return authSession;
 }
 
++ (NSArray<MXLoginFlow*>*)loginFlowsFromJSON:(NSArray *)JSONDictionaries
+{
+    NSMutableArray *loginFlows;
+    
+    for (NSDictionary *JSONDictionary in JSONDictionaries)
+    {
+        MXLoginFlow *loginFlow;
+        
+        NSString *type;
+        
+        MXJSONModelSetString(type, JSONDictionary[@"type"]);
```

Comment:
I'm not very comfortable with exposing the name of `type` field outside of `MXLoginFlow` class. Maybe we can define an extern variable for that.

Reply:
Sure

Reply:
This should be avoided after https://github.com/matrix-org/matrix-ios-sdk/pull/965#discussion_r536024589